kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyExceptionHandler.java

```diff
@@ -196,11 +199,20 @@ public void registerExceptionResponse(Class<? extends Exception> exceptionClass,
     @SuppressWarnings("java:S1452")
     public Optional<?> handleException(Throwable throwable) {
         var localCause = throwable;
+        Set<Throwable> visitedExceptions = Collections.newSetFromMap(new IdentityHashMap<>());
```

Comment:
@k-wall pointed at LOGBACK-1027 which when I followed up had this comment https://github.com/qos-ch/logback/pull/399#issuecomment-639565260 so dealing full loops seemed appropriate. No idea when it would actually make sense but they are constructible in tests (if contrived).